packages/front-end/components/SchemaBrowser/EditSqlModal.tsx

```diff
@@ -54,19 +68,110 @@ export default function EditSqlModal({
+      // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
+      const userIdTypes = identityTypes.map((type) => type.userIdType || []);
+
+      const returnedColumns = new Set<string>(Object.keys(result));
+      const optionalColumns = [...returnedColumns].filter(
+        (col) =>
+          !requiredColumns.has(col) &&
+          !namedCols.includes(col) &&
+          !userIdTypes.includes(col)
+      );
+
+      // Check if `hasNameCol` should be enabled
+      if (!userEnteredHasNameCol) {
+        // Selected both required columns, turn on `hasNameCol` automatically
+        if (
+          returnedColumns.has("experiment_name") &&
+          returnedColumns.has("variation_name")
+        ) {
+          setHasNameCols(true);
+        }
+        // Only selected `experiment_name`, add warning
+        else if (returnedColumns.has("experiment_name")) {
+          suggestions.push(
+            <>
+              Add <code>variation_name</code> to your SELECT clause to enable
+              GrowthBook to populate names automatically.
+            </>
+          );
+        }
+        // Only selected `variation_name`, add warning
+        else if (returnedColumns.has("variation_name")) {
+          suggestions.push(
+            <>
+              Add <code>experiment_name</code> to your SELECT clause to enable
+              GrowthBook to populate names automatically.
+            </>
+          );
+        }
+      }
+
+      // Prompt to add optional columns as dimensions
+      if (optionalColumns.length > 0) {
+        suggestions.push(
+          <>
+            The following columns were returned, but will be ignored. Add them
+            as dimensions or disregard this message.
+            <ul className="mb-0 pb-0">
+              {optionalColumns.map((col) => (
+                <li key={col}>
+                  <code>{col}</code> -{" "}
+                  <a
+                    href="#"
+                    onClick={(e) => {
+                      e.preventDefault();
+                      setDimensions([...userEnteredDimensions, col]);
+                    }}
+                  >
+                    add as dimension
+                  </a>
+                </li>
+              ))}
+            </ul>
+          </>
+        );
+      }
+
+      setSuggestions(suggestions);
+    }
+  }, [
+    requiredColumns,
+    testQueryResults,
+    form,
+    queryType,
+    userEnteredHasNameCol,
+    userEnteredDimensions,
+    identityTypes,
+    setHasNameCols,
+    setDimensions,
+  ]);
+
+  const handleTestQuery = async () => {
     const sql = form.getValues("sql");
+    setTestQueryResults(null);
     try {
-      // Just check for basic SQL syntax, not any required columns
-      // We can check for required columns after the results are returned
-      validateSQL(sql, []);
-
+      validateSQL(sql, [...requiredColumns]);
```

Comment:
Do we want to change this?  The way it was before, someone could run a quick `SELECT * FROM ...` just to get a sense of what the data looked like before crafting the full query.

With this change, we are now doing super strict validation and blocking them unless they explicitly select every required column.  This could be frustrating to users.